### DISCUSSION TITLE
Entity Level Sentiment Analysis or Targeted Sentiment Analysis

### DISCUSSION THREAD

**Discussion by shrinidhin** [explosion/spaCy#10113] [Help: Coding & Implementations] [answered]
Hello!I am working on a project which processes news articles and gives information on whether the news is positive or negative for different entities mentioned in the news article. I have used spacy and built a custom NER model for recognizing the entities. i am currently working on the sentiment analyzer bit. Consider the following eg:

`Indian shares flitted between gains and losses on Thursday, with weakness in Wipro and Tata Motors countering robust moves in technology stocks after Infosys and Tata Consultancy Services reported strong third-quarter results. The NSE Nifty 50 index (.NSEI) rose 0.03% to 18,215.55 by 0449 GMT, while the S&P BSE Sensex (.BSESN) added 0.01% to 61,147.87. Still, both the indexes are up roughly 5% this year so far."We had a good run in the lead up to IT results, and investors are now reviewing things; as more earnings come in, moves are going to be reactionary instead of (being) proactive," said Anand James, chief market strategist at Geojit Financial Services.Tech giants Infosys (INFY.NS) and Tata Consultancy Services (TCS.NS) jumped around 1.5% each. Infosys on Wednesday lifted its revenue outlook for the financial year 2022, while TCS reported a 12.3% rise in quarterly profit.However, rival Wipro slid 5.5%`

So the entities detected will be `Wipro, Tata Motors, Infosys and Tata Consultancy Services.` I was thinking of now extracting the sentences which give information about each entity, and then use them as inputs for the sentiment analysis model, which gives results as Positive or Negative for each entity. 
I tried using dependency parsing but it hasn't been very effective in picking up phrases for each entity where the nsubj doesn't have the entity names explicitly. 
I did find [this answer](https://github.com/explosion/spaCy/discussions/8931) in the discussions with a similar problem statement. However I am yet to understand how to extract the representative statements for each entity. I would really appreciate any inputs!Thanks!

**Comment by thomashacker** [comment, marked as answer]
Hello,
We've recently published a blog post about [Healthsea](https://explosion.ai/blog/healthsea), a pipeline that analyzes reviews to supplements that tries to solve the same problem. If you're interested you can have a look, it describes every processing step from extracting sentences to predicting their sentiment.

**Reply by shrinidhin** [reply]
Awesome!Thanks a lot!

**Reply by shrinidhin** [reply]
I would like to understand why  a textcat component is used. Since sentiment analysis is the goal here, wouldn't it be better to use something like textblob instead of textcat for this?Even though sentiment analysis can be modeled as a text classification problem here.

**Reply by thomashacker** [reply]
I haven't worked with textblob before, but a good advantage of using `textcat` is that the sentiment analysis will be trained on your specific data, rather than an algorithmic approach or a pre-trained model which might give worse results. However, feel free to experiment with both and see which one works best for your usecase.

**Reply by shrinidhin** [reply]
Alright. Thank you!
